### PR TITLE
Feat/assign orders logic

### PR DESCRIPTION
**Pull request by Parle648**
## Describe your changes

- In this PR, I moved the logic for assigning orders to drivers from the front-end to the back-end. Additionally, I added error handling to the algorithm for cases where orders have the same collection_time_start value. I also improved the validation for creating routes: now, orders with different dates cannot be selected, and the number of selected drivers cannot exceed the number of selected orders

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-119)

## Loom video
https://www.loom.com/share/bcdaea1c435e4f139d83ed84a8a40ecd?sid=7debaf47-6098-4214-8e65-daf127f35d11

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { createTimeRange } from '@/utils/createTimeRange';
+import { RootState } from '@/store/store';
+import { errorSubtitleStyles, errorTitleStyle } from './styles';
+
+const UnassignedOrder: FC = () => {
+  const { notAssignedOrders } = useSelector(
+    (store: RootState) => store.ordersToDriversSlice
+  );
+
+  const { setNodeRef } = useDroppable({ id: -1 });
```
Comment (AlexeyBezrebriy) [inline src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx:17]: Magic numbers

Comment (AlexeyBezrebriy) [inline src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx]: Magic numbers

File: src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            <OrderRow
+              key={order.id}
+              order={{
+                ...order,
+                time_range: createTimeRange(
+                  order.collection_time_start,
+                  order.collection_time_end
+                ),
+                city: order.collection_address?.split(',')[
+                  order.collection_address.split(',').length - 2
```
Comment (AlexeyBezrebriy) [inline src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx:39]: Magic numbers

File: src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                ],
+              }}
+              parentId={-1}
```
Comment (AlexeyBezrebriy) [inline src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx]: Magic numbers

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx (hunk 1 above)
Comment (Parle648) [reply src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx]: done

File: src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx (hunk 3 above)
Comment (Parle648) [reply src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx]: done

**Review by oyakovytskyi — APPROVED** [review]

File: src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx (hunk 2 above)
Comment (lyudad) [reply src/pages/CreateRoute/RouteManagement/components/InformBlock/components/UnassignedOrder/index.tsx:39]: no its fine

**Review by lyudad — APPROVED** [review]

**Pull request merged**